src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -338,7 +346,7 @@ public Territory retreatQuery(final GUID battleID, final boolean submerge, final
         && (battleTerritory.isWater() || Match.someMatch(attackers, Matches.UnitIsLand))) {
       return null;
     }
-    ProBattleUtils.setData(getGameData());
+    calc.setData(getGameData());
```

Comment:
Again, what uses the calc that required this set. Why isn't hte usage that required it and this set closer together? Seems magic that we'll just so happen ot have the right data state in the cacl when we use it next. Another word for magic is error prone and brittle..